### PR TITLE
Fixes #24313 - load data to switcher in smart class parameters

### PR DESCRIPTION
**Pull request by lizagilman**
The error occurs because the expected url does not match the default url template. Added adaptations to resolve it.
/cc @amirfefer

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js [hunk 1]
```diff
@@ -44,11 +44,14 @@ export const loadSwitcherResourcesByResource = (resource, { page = 1, searchQuer
...
-      name: get(result, nameField),
-      id: result.id,
-      url: switcherItemUrl.replace(':id', result.id),
-    }));
+    const switcherItems = flatten(Object.values(data.results)).map((result) => {
+      const itemName = get(result, nameField);
+      return ({
+        name: get(result, nameField),
+        id: result.id,
+        url: switcherItemUrl.replace(':id', result.id).replace(':name', itemName),
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js]: instead calling `replace` twice, we can do something like this:
```javascript
const mapTemplateVariable = {
   ':id': result.id,
   ':name': itemName,
};
return ({
  url: switcherItemUrl.replace(/:id|:name/gi, (matched) =>  mapTemplateVariable[matched]);
})
```
With that we can extend item url template with ease later on.

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js [hunk 2]
```diff
@@ -44,11 +44,14 @@ export const loadSwitcherResourcesByResource = (resource, { page = 1, searchQuer
...
   const formatResults = ({ data }) => {
-    const switcherItems = flatten(Object.values(data.results)).map(result => ({
-      name: get(result, nameField),
-      id: result.id,
-      url: switcherItemUrl.replace(':id', result.id),
-    }));
+    const switcherItems = flatten(Object.values(data.results)).map((result) => {
+      const itemName = get(result, nameField);
+      return ({
+        name: get(result, nameField),
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js]: you can use `itemName` from above

**Review by amirfefer — CHANGES_REQUESTED** [review]
nice work @lizagilman :+1: 
Just few comments :)

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js (hunk 1 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js]: IMHO using `replace` looks better for readability and easier to understand.

Comment (lizagilman) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js]: @sharvit maybe leave it more readable now, and when we grow with the number of template variables we"ll change to a mapper?

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js (hunk 2 above)
Comment (lizagilman) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBarActions.js]: oops :) updating

**Review by sharvit — APPROVED** [review]
LGTM! Thanks @lizagilman :+1:

**Review by dLobatog — APPROVED** [review]
Nice, thanks @lizagilman - this would be very useful in https://github.com/theforeman/foreman_ansible/pull/193 in case you want to take a stab at it

**Pull request merged**